src/plugins/intel_gna/src/ops/util/util.cpp

```diff
@@ -0,0 +1,165 @@
+                       bool is_exception_allowed) {
+    if (!conv_ie) {
+        if (is_exception_allowed) {
+            THROW_GNA_EXCEPTION << "ConvolutionIE node is empty!\n";
+        }
+        return false;
+    }
+    size_t batch_size = conv_ie->input_value(0).get_shape()[0];
+    if (batch_size != 1) {
+        if (is_exception_allowed) {
+            THROW_GNA_EXCEPTION << "topology with layer: " + conv_ie->get_friendly_name() + ", type: " + conv_ie->get_type_name()  +
+            ", and batch size(" + std::to_string(batch_size) + ") != 1 not supported";
+        }
+        return false;
+    }
+    auto check_dilation = [&](size_t filter_dilation_height, size_t filter_stride_width) -> bool {
+        Cnn2D::RangeLimit2D dilation_limit{
+            {convDilationHeight, convDilationHeight, "dilation height"},
+            {convDilationWidth, convDilationWidth, "dilation width"}};
+        std::string error = dilation_limit.GetErrorOrEmpty(filter_dilation_height, filter_stride_width);
+        return Cnn2D::AbstractValidator::ValidationSuccesful(is_exception_allowed, error, conv_ie->get_friendly_name(), conv_ie->get_type_name());
+    };
+    auto input_shape = conv_ie->input_value(0).get_shape();
+    auto filter_shape = conv_ie->input_value(1).get_shape();
+    if ((4 == filter_shape.size() && filter_shape[2] > 1 && filter_shape[3] > 1) ||
+        (4 == input_shape.size() && input_shape[2] > 1 && input_shape[3] > 1)) {
+        pass::helper::ConvData conv_data;
+        pass::helper::UpdateConvData(conv_ie, conv_data);
+        if (GNAPluginNS::GNAConvolutionLayer::isMappableFrom2DTo1D(
+            conv_data.input_height, conv_data.input_width, conv_data.input_channel_count,
+            conv_data.filter_height, conv_data.filter_width,
+            conv_data.filter_stride_height, conv_data.filter_stride_width)) {
+            return check_dilation(conv_data.filter_dilation_height, conv_data.filter_dilation_width);
+        }
+        const auto cnn2dValidatorPtr = Cnn2D::AbstractValidator::Create(gnaCompileTarget);
+        if (cnn2dValidatorPtr) {
+            return cnn2dValidatorPtr->ValidateCnn2D(conv_ie->get_friendly_name(),
+                conv_data.input_height, conv_data.input_width, conv_data.input_channel_count,
+                conv_data.filter_height, conv_data.filter_width, conv_data.filter_channel_count,
+                conv_data.filter_stride_height, conv_data.filter_stride_width,
+                conv_data.filter_dilation_height, conv_data.filter_dilation_width,
+                OvGnaTypeIntFromBytes(config.gnaPrecision.size()), is_exception_allowed);
+        }
+    }
+    return check_dilation(conv_ie->get_dilations()[0], conv_ie->get_dilations()[1]);
+}
+
+bool is_pooling_supported(const std::shared_ptr<ngraph::opset7::MaxPool> max_pool,
+                          const std::string& gnaCompileTarget,
+                          bool is_exception_allowed) {
+    if (!max_pool) {
```

Comment:
I just wonder if there is some desynchronization between ngraph and legacy layers handlers.

I see a lot of magic numbers and I just wonder if we should have some namespace or class which would contain declaration of limits/configuration of limits. It would easier to update later and spot issues.

Reply:
> I just wonder if there is some desynchronization between ngraph and legacy layers handlers.
> 
> I see a lot of magic numbers and I just wonder if we should have some namespace or class which would contain declaration of limits/configuration of limits. It would easier to update later and spot issues.

I tried to reuse limitations where it is possible.